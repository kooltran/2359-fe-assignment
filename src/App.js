import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { getBarAction } from "./redux/actions/getBarAction";
import BarDropDown from "./component/barDropdown";
import ProgressBarCmp from "./component/progressBar";
import ProgressButton from "./component/progressButton";
import loadingIcon from "./assets/images/loading.svg";

export class App extends React.Component {
  static propTypes = {
    data: PropTypes.shape({
      bars: PropTypes.array.isRequired,
      buttons: PropTypes.array.isRequired,
      limit: PropTypes.number.isRequired
    }),
    fetchState: PropTypes.string.isRequired,
    getBarAction: PropTypes.func.isRequired
  };

  componentDidMount() {
    this.props.getBarAction();
  }

  render() {
    const { data, fetchState } = this.props;
    return (
      <div className="container">
        {fetchState === "loading" && (
          <div className="progress-loading">
            <img src={loadingIcon} alt="loading" />
          </div>
        )}
        {fetchState === "success" && (
          <div>
            <h1>Limited value: {data.limit}</h1>
            <BarDropDown />
            <ProgressBarCmp bars={data.bars} limit={data.limit} />
            <ProgressButton buttons={data.buttons} />
          </div>
        )}
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    data: state.barsInfoReducer.data,
    fetchState: state.barsInfoReducer.fetchState
  };
};

const mapDispatchToProps = {
  getBarAction
};

export default connect(mapStateToProps, mapDispatchToProps)(App);
