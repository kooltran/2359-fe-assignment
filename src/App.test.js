import React from "react";
import { shallow } from "enzyme";
import { App } from "./App";

describe("renders App component", () => {
  const baseProps = {
    fetchState: "loading",
    getBarAction: jest.fn()
  };

  it("should render properly with loading state", () => {
    expect(shallow(<App {...baseProps} />)).toMatchSnapshot();
  });

  it("should dispatch an action on componentDidMount", () => {
    expect(baseProps.getBarAction).toHaveBeenCalledTimes(1);
  });
});
