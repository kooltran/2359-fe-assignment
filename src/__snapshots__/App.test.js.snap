// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App component should render properly with loading state 1`] = `
<div
  className="container"
>
  <div
    className="progress-loading"
  >
    <img
      alt="loading"
      src="loading.svg"
    />
  </div>
</div>
`;
